dds/DCPS/XTypes/MemberDescriptorImpl.cpp

```diff
@@ -89,7 +154,7 @@ bool test_equality(DDS::MemberDescriptor* lhs, DDS::MemberDescriptor* rhs, Dynam
     std::strcmp(lhs->name(), rhs->name()) == 0 &&
     lhs->id() == rhs->id() &&
     test_equality(lhs_type, rhs_type, dt_ptr_pair) &&
-    lhs->default_value() == rhs->default_value() &&
+    std::strcmp(lhs->default_value(), rhs->default_value()) == 0 &&
```

Comment:
This is probably okay, but doesn't seem like it's technically correct since there could be multiple string representations for some values, eg `0x10` and `16`.